go/vt/vttablet/tabletmanager/vreplication/controller_plan_test.go

```diff
@@ -85,16 +101,33 @@ func TestControllerPlan(t *testing.T) {
 		// Update
 	}, {
 		in: "update _vt.vreplication set state='Running' where id = 1",
-		plan: &controllerPlan{
-			opcode: updateQuery,
-			query:  "update _vt.vreplication set state = 'Running' where id = 1",
-			id:     1,
+		plan: &testControllerPlan{
+			query:    "update _vt.vreplication set state='Running' where id = 1",
+			opcode:   updateQuery,
+			selector: "select id from _vt.vreplication where id = 1",
+			applier:  "update _vt.vreplication set state = 'Running' where id in ::ids",
+		},
+	}, {
+		in: "update _vt.vreplication set state='Running'",
+		plan: &testControllerPlan{
+			query:    "update _vt.vreplication set state='Running'",
+			opcode:   updateQuery,
+			selector: "select id from _vt.vreplication",
+			applier:  "update _vt.vreplication set state = 'Running' where id in ::ids",
+		},
+	}, {
+		in: "update _vt.vreplication set state='Running' where a = 1",
+		plan: &testControllerPlan{
+			query:    "update _vt.vreplication set state='Running' where a = 1",
+			opcode:   updateQuery,
+			selector: "select id from _vt.vreplication where a = 1",
+			applier:  "update _vt.vreplication set state = 'Running' where id in ::ids",
 		},
 	}, {
 		in: "update _vt.resharding_journal set col = 1",
-		plan: &controllerPlan{
-			opcode: reshardingJournalQuery,
+		plan: &testControllerPlan{
 			query:  "update _vt.resharding_journal set col = 1",
+			opcode: reshardingJournalQuery,
 		},
 	}, {
 		in:  "update a set state='Running' where id = 1",
```

Comment:
it will be good to change `a` to `_vt.vreplication` here unless you deliberately want to test with a different table name

Reply:
Yeah. This test shows that `a` will not be accepted as a valid table name.